lib/internal/process.js

```diff
@@ -60,7 +60,7 @@ function setupConfig(_source) {
       .replace(/"/g, '\\"')
       .replace(/'/g, '"');
 
-  process.config = JSON.parse(config, function(key, value) {
+  process.config = exports.config = JSON.parse(config, (key, value) => {
```

Comment:
Yep. Modification is still possible. The problem we're having in userland is with code that does stuff like `process.config = {}`, essentially overwriting everything. The userland code that extends the object hasn't been a problem that I can determine. This is an attempt at doing the simplest thing first. If we need to lock it down more, then let's do so.